src/execution/execute.js

```diff
@@ -780,9 +823,17 @@ function completeListValue(
   // where the list contains no Promises by avoiding creating another Promise.
   const itemType = returnType.ofType;
   let containsPromise = false;
-  const completedResults = result.map(item => {
-    const completedItem =
-      completeValueCatchingError(exeContext, itemType, fieldASTs, info, item);
+  const completedResults = result.map((item, index) => {
+    const childExePath = exePath.slice();
+    childExePath.push(index);
+
+    const childInfo = Object.assign({}, info, {
```

Comment:
Ah yes, I see now why you're correct. Child fields will still rely on this portion of the path to be added. My hope was that this intermediate `childInfo` object should not need to be constructed since it is never provided to a resolver function (instead, it is provided to the `resolveType` and `isTypeOf` functions) - Perhaps only the `childExePath` is necessary but the new `childInfo` is not?